delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,292 @@
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Create a new resolver with the provided id
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(id: Iri, project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Update an existing resolver
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param rev            the current revision of the resolver
+    * @param resolverFields the payload to update the resolver
+    */
+  def update(id: Iri, project: ProjectRef, rev: Long, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Add a tag to an existing resolver
+    *
+    * @param id        the resolver identifier
+    * @param project   the project where the resolver belongs
+    * @param tag       the tag name
+    * @param tagRev    the tag revision
+    * @param rev       the current revision of the resolver
+    */
+  def tag(id: Iri, project: ProjectRef, tag: Label, tagRev: Long, rev: Long)(implicit
+      caller: Subject
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Deprecate an existing resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver belongs
+    * @param rev     the current revision of the resolver
+    */
+  def deprecate(id: Iri, project: ProjectRef, rev: Long)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Fetch the last version of a resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver belongs
+    */
+  def fetch(id: Iri, project: ProjectRef): UIO[Option[ResolverResource]]
+
+  /**
+    * Fetches the resolver at a given revision
+    * @param id      the resolver identifier
+    * @param project the project where the resolver belongs
+    * @param rev     the current revision of the resolver
+    */
+  def fetchAt(id: Iri, project: ProjectRef, rev: Long): IO[RevisionNotFound, Option[ResolverResource]]
+
+  /**
+    * Fetches a resolver by tag.
+    *
+    * @param id        the resolver identifier
+    * @param project   the project where the resolver belongs
+    * @param tag       the tag revision
+    */
+  def fetchBy(
+      id: Iri,
+      project: ProjectRef,
+      tag: Label
+  ): IO[ResolverRejection, Option[ResolverResource]]
```

Comment:
fetchBy can be implemented directly here since you have all necessary bits:
```scala
fetch(id, projectRef).flatMap {
      case Some(resource) =>
        resource.value.tags.get(tag) match {
          case Some(rev) => fetchAt(id, projectRef, rev).leftMap(_ => TagNotFound(tag))
          case None      => IO.raiseError(TagNotFound(tag))
        }
      case None           => IO.pure(None)
    }
```

Reply:
Well, I thought it wasn't in the scope of this PR